final-progII/src/persistence/ControllerDB.java

```diff
@@ -716,4 +715,51 @@ public void deleteAbsence(Absence absence) throws Exception {
 
 		return absences;
 	}
+
+	public List<Subject> recoverTeaches(int ci) throws Exception {
+		List<Subject> teaches = new ArrayList<Subject>();
+
+		try {
+
+			this.MySQLconnection();
+
+			System.out.println("Creating a connection Object");
+			PreparedStatement teachesSt = this.conn.prepareStatement("SELECT * FROM teaches WHERE ciTeacher = ?");
+
+			teachesSt.setInt(1, ci);
+
+			System.out.println("Executing Query and creating ResultSet");
+			ResultSet teachesRs = teachesSt.executeQuery();
+
+			while (teachesRs.next()) {
+				teaches.add(new Subject(teachesRs.getString("idSubject")));
+			}
+
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+		return teaches;
+
+	}
+
+	public void deleteTeacherFromTeaches(int id) throws Exception {
+
+		try {
+			System.out.println("Creating a connection");
+			this.MySQLconnection();
+
+			System.out.println("Creating a connection Object");
+			PreparedStatement deleteTeachesSt = this.conn.prepareStatement("DELETE FROM teaches WHERE idSubject = ?");
```

Comment:
This method would delete a subject related to a teacher, and it's currently not working in that way.
Maybe you could add a parameter to your query that would be the teacher CI.
You can see this error if you test it with more than one theacher.

Reply:
Okey, I'll fix it